docs/architecture/blazor-for-web-forms-developers/migration.md

```diff
@@ -8,8 +8,322 @@ ms.date: 09/11/2019
+
+- Configure services required in the application
+  
+  Dependency injection is built into ASP.NET Core from the bottom up so
+  everything can be considered a service that can be scoped how needed. This can
+  be per-instance, per-request, or an application singleton. This startup method
+  allows you to add services for your own application, as well as override
+  system ones with your own implementation.
+
+- Configure pipeline of middleware
+
+  Middleware can be thought of as handlers that take a request in and has the
+  ability to process it before or after sending it on to the next piece of
+  middleware. There are many built in middleware that can be added into the
+  pipeline, as well as easy ways to add your own implementations to do things
+  like inspect or add headers, exception handling, logging, and many more. This
+  will be discussed more in depth in the chapter on
+  [Middleware](./middleware.md).
+
+## Example of code changes
+
+We'll demonstrate some of these code changes by looking at the [eShopOnBlazor](https://github.com/dotnet-architecture/eShopOnBlazor) project. This project is a simple eShop that allows a user to perform simple CRUD operations on a catalog. We will look specifically at the Create page and the changes required to make that work.
+
+### Start up changes
+
+In Blazor, the startup process is the same as other ASP.NET Core frameworks.
+This is greatly simplified from traditional ASP.NET where everything that is
+included is in the start up class itself. Comparing `Global.asax.cs` vs
+`Startup.cs`, the separation of the start up into services and pipeline make it
+clear what is being registered. In WebForms, dependency injection was not
+something that was built into the project, but something you could add on. For
+projects that wanted to use it, it was difficult to get things hooked up right,
+as can be seen in `Global.asax.cs` as the container has to be wired up
+manually.This is done for you in ASP.NET Core. Anything that is required for
+startup that had been in `Global.asax` can be added into this file to configure
+the ASP.NET pipeline.
+
+### Code behind vs razor page
+
+In WebForms, a page was defined by three files: Create.aspx,
+Create.aspx.designer.cs and Create.aspx.cs. In Blazor, it is all done in the
+same file, with any code generation from the HTML elements occurring behind the
+scenes (equivalent to the `*designer.cs` files).
+
+The first thing that you will see is that the syntax is slightly different from
+WebForms to Razor. WebForms components are brought in by a tag similar to
+`<asp:SomeComponent>` while in Blazor this will look like `<SomeComponent>`. The
+Razor compiler will understand that this is a Razor component and will handle it
+appropriately. The syntax to start injecting code is different, too; instead of
```

Comment:
Razor syntax is covered in the components doc. It's ok to reiterate this stuff as appropriate, but as much as possible we should should focus on migration steps here.